java/core/src/main/java/sleeper/core/statestore/AllFileReferences.java

```diff
@@ -54,20 +82,53 @@ public boolean equals(Object o) {
             return false;
         }
         AllFileReferences that = (AllFileReferences) o;
-        return moreThanMax == that.moreThanMax && Objects.equals(filesWithNoReferences, that.filesWithNoReferences) && Objects.equals(activeFiles, that.activeFiles);
+        return moreThanMax == that.moreThanMax && Objects.equals(filesByFilename, that.filesByFilename);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(filesWithNoReferences, activeFiles, moreThanMax);
+        return Objects.hash(filesByFilename, moreThanMax);
     }
 
     @Override
     public String toString() {
         return "AllFileReferences{" +
-                "filesWithNoReferences=" + filesWithNoReferences +
-                ", activeFiles=" + activeFiles +
+                "filesByFilename=" + filesByFilename +
                 ", moreThanMax=" + moreThanMax +
                 '}';
     }
+
+    private static Map<String, FileReference> activeFilesByFilenameAndPartition(Stream<FileReference> activeFiles) {
+        Map<String, FileReference> map = new LinkedHashMap<>();
+        activeFiles.forEach(file -> map.put(file.getFilename() + "|" + file.getPartitionId(), file));
```

Comment:
In the state store we use partitionId followed by filename, might be worth keeping that consistency?